test/c/test_circuit.c

```diff
@@ -932,6 +935,26 @@ static int test_delay_instruction(void) {
     return result;
 }
 
+/**
+ * Test dag to circuit conversion.
+ */
+static int test_circuit_to_dag(void) {
+    QkCircuit *qc = qk_circuit_new(2, 1);
+
+    QkDag *dag = qk_circuit_to_dag(qc, false, (uint32_t[]){1, 0}, NULL);
+
+    int result = Ok;
+    if (qk_dag_num_qubits(dag) != 2 || qk_dag_num_clbits(dag) != 1) {
```

Comment:
Not for this PR, but maybe we ought to have a `qk_dag_is_equal` function as part of our P2 items, specifically to make our C testing more robust.